python/ray/train/gbdt_trainer.py

```diff
@@ -297,6 +304,15 @@ def training_loop(self) -> None:
 
         config[self._init_model_arg_name] = init_model
 
+        if init_model:
+            # If restoring, make sure that we only create num_boosting_round trees,
+            # and not init_model_trees + num_boosting_round trees
+            last_iteration = self._model_iteration(init_model)
+            num_iterations = config.get(
+                self._num_iterations_argument, self._default_num_iterations
+            )
+            config[self._num_iterations_argument] = num_iterations - last_iteration
```

Comment:
Hmm, now I'm thinking `restore` is completely the same as `resume_from_checkpoint` for xgboost.

I thought:
- `resume_from_checkpoint` = start a *new* training run, initializing from a checkpoint ("fine-tuning" for `num_boost_round` more iters), not using the logic here.
- `restore` = continue an *old* training run, logging to the same dir, and continuing until `num_boost_round` as specified from the original configs (using the logic here)

Reply:
We can probably distinguish between the two with `trainer._restore_path`. What do you think?

Reply:
It's tricky, because if the Trial is restarted during Experiment, it will use the `resume_from_checkpoint` path, even though we want it to just continue to `num_boost_round` trees instead of fitting `num_boost_round` more. So in other words, the only situation where we may want to keep the old logic is when the user themselves passes the checkpoint in `resume_from_checkpoint`, but that may be tricky to determine.

Reply:
How about this: We treat them the same, i.e. whatever is passed to `num_boost_rounds` is how many _final_ iterations the model should have been trained for.

E.g. when a user passes a model that has been trained for 20 iterations, with num_boost_round=30, it is trained for 10 more iterations and then stops.

We make this very clear in the docstring. We can also log an info message if we adjusted the num boost rounds to make it even clearer. We throw an error if the num_boost_rounds is smaller than what the model has already been trained for.

Then we don't have to maintain multiple complex paths. If users want to continue training an existing model for n more rounds, they just have to adjust n themselves.

Reply:
Yeah, I think this would be the best way forward.

Reply:
Yes, happy with this!